grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaChannel.java

```diff
@@ -62,23 +68,31 @@
     private final MeterRegistry meterRegistry;
     private final SessionProtocol sessionProtocol;
     private final SerializationFormat serializationFormat;
-    @Nullable
-    private final MessageMarshaller jsonMarshaller;
+    private final GrpcMessageMarshaller messageMarshaller;
     private final String advertisedEncodingsHeader;
 
     ArmeriaChannel(ClientBuilderParams params,
                    HttpClient httpClient,
                    MeterRegistry meterRegistry,
                    SessionProtocol sessionProtocol,
                    SerializationFormat serializationFormat,
+                   GrpcMessageMarshaller messageMarshaller,
                    @Nullable MessageMarshaller jsonMarshaller) {
         this.params = params;
         this.httpClient = httpClient;
         this.meterRegistry = meterRegistry;
         this.sessionProtocol = sessionProtocol;
         this.serializationFormat = serializationFormat;
-        this.jsonMarshaller = jsonMarshaller;
-
+        if (messageMarshaller != NoopGrpcMessageMarshaller.get()) {
```

Comment:
Maybe better using `null` instead of `NoopGrpcMessageMarshaller`?

Reply:
I tried at first, but I realized a `ClientOption` does not take `null` as a default value.
```java
// Will throw NullPointerException
ClientOption.define("GRPC_MESSAGE_MARSHALLER", null);
```
https://github.com/line/armeria/blob/56c58ffb78c7dc7d6845fb08653c0503b78ce4b0/core/src/main/java/com/linecorp/armeria/common/util/AbstractOption.java#L113-L118

Probably better to make it accept null?

Reply:
That's a good point. If we accept `null`, we will have to add `@Nullable` to the getter methods which doesn't sound like a good idea. Let's leave this as it is. :sweat_smile:

Reply:
Agreed!